packages/desktop-client/src/components/schedules/DiscoverSchedules.tsx

```diff
@@ -110,33 +123,50 @@ function DiscoverSchedulesTable({ schedules, loading }) {
   );
 }
 
-export default function DiscoverSchedules({ modalProps, actions }) {
-  let { data: schedules, isLoading } =
-    useSendPlatformRequest('schedule/discover');
-  if (!schedules) schedules = [];
-
-  let [creating, setCreating] = useState(false);
-
-  let selectedInst = useSelected('discover-schedules', schedules, []);
+export default function DiscoverSchedules({
+  modalProps,
+  actions,
+}: {
+  modalProps: CommonModalProps;
+  actions: BoundActions;
+}) {
+  const { data, isLoading } = useSendPlatformRequest('schedule/discover') as {
+    data: Awaited<ReturnType<SchedulesHandlers['schedule/discover']>> | null;
+    isLoading: boolean;
+  };
+
+  const schedules = data || [];
+
+  const [creating, setCreating] = useState(false);
+
+  const selectedInst = useSelected<DiscoverScheduleEntity>(
+    'discover-schedules',
+    schedules,
+    [],
+  );
 
   async function onCreate() {
-    let selected = schedules.filter(s => selectedInst.items.has(s.id));
+    const selected = schedules.filter(s => selectedInst.items.has(s.id));
     setCreating(true);
 
     for (let schedule of selected) {
-      let scheduleId = await send('schedule/create', {
+      const scheduleId = await send('schedule/create', {
         conditions: schedule._conditions,
+        schedule: {},
       });
 
       // Now query for matching transactions and link them automatically
-      let { filters } = await send('make-filters-from-conditions', {
+      const { filters } = await send('make-filters-from-conditions', {
         conditions: schedule._conditions,
       });
 
       if (filters.length > 0) {
-        let { data: transactions } = await runQuery(
+        const {
+          data: transactions,
+        }: { data: Pick<TransactionEntity, 'id'>[] } = await runQuery(
```

Comment:
<!-- Warning represent user-facing problems. If possible, it's great to follow this kind of comment with a suggestion.  -->
**🔨 warning:** ‏we should be also able to automatically infer the type here.
```suggestion
        } = await runQuery(
```

Reply:
This doesn't work now., However I removed the implicit typing, 
This needs type enforcement in `q` func